monitoring/reporter/reporter.go

```diff
@@ -0,0 +1,32 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package reporter
+
+import (
+	"github.com/elastic/elastic-agent-libs/opt"
+)
+
+// The reporter interface allows for the creation of new metrics outputs that can sent metrics to any user-configured output. Expvar, logs, etc
+
+// QueueMetrics represents the queue health metrics to be reported to the user, the queue.Metrics struct is "closer to the code", while this is meant to represent useful and human-readable metrics in a stable interface.
+// We're keeping this as a struct to prevent the metrics from becoming a free-for-all; the values here should be as self-documenting as possible, and give users quick insight into the state of the queue.
+type QueueMetrics struct {
+	//QueueLimitReachedCount is the number of times the queue has reached its user-configured limit.
+	QueueLimitReachedCount opt.Uint `struct:"queue_limit_reached_count,omitempty" json:"queue_limit_reached_count"`
+	//QueueIsCurrentlyFull reports if the queue is currently at its user-configured limit
+	QueueIsCurrentlyFull bool `struct:"queue_is_currently_full,omitempty" json:"queue_is_currently_full"`
+	//CurrentQueueLevel reports the current fill state of the queue, in the native user-configured limits of the queue
+	CurrentQueueLevel opt.Uint `struct:"current_queue_level,omitempty" json:"current_queue_level"`
```

Comment:
[Question | Suggestion]
It's my understanding that these metrics are already the current state of the queue, therefore I find it a bit redundant to have _current/currently_ for only these 2 metrics. Besides the other metrics do not have it. What could be understood as they being not the current state.

```suggestion
	// Full reports if the queue is currently at its user-configured limit
	 Full bool `struct:"queue_is_currently_full,omitempty" json:"queue_is_currently_full"`
	// Level reports the current fill state of the queue, in the native user-configured limits of the queue
	Level opt.Uint `struct:"current_queue_level,omitempty" json:"current_queue_level"`
```